cardinal/world.go

```diff
@@ -207,6 +208,35 @@ func NewMockWorld(opts ...WorldOption) (*World, error) {
 	return world, nil
 }
 
+func RegisterMessage[In any, Out any](world *World, name string, opts ...message.MessageOption[In, Out]) error {
+	msgType := message.NewMessageType[In, Out](name, opts...)
+	err := world.registerMessagesByName(msgType)
+	if err != nil {
+		return eris.Wrap(err, "failed to register message")
+	}
+	typeValueOfMessageType := reflect.TypeOf(*msgType)
+	return world.GetMessageManager().RegisterMessageByType(typeValueOfMessageType, msgType)
+}
+
+func (w *World) registerMessagesByName(msgs ...types.Message) error {
+	return w.msgManager.RegisterMessagesByName(msgs...)
+}
+
+func GetMessageFromWorld[In any, Out any](world *World) (*message.MessageType[In, Out], error) {
```

Comment:
I'm confident that`GetMessageFromWorld` and `GetMessage` should not exist; this looks fine now, but then everyone wants to code and merge their PR quickly, people also review PR sloppily, and this sooner or later this will just be a nest of tech debt. 

I saw that you added this only for the `TestCannotWaitForNextTickAfterEngineIsShutDown` test, which honestly a good showcase the pile of tech debt the codebase is dealing with.

`AddEVMTransaction` violates abstraction boundaries by operating with `ID` that should have been for internal records only. If it just operates on the string name identifier, then you wouldn't have this issue in the first place because you name in advance what the name of the message is going to be.

Additionally, there is just a larger problem with tests in this codebase in general, because everyone wants to integration tests, every time they want to violate abstraction boundaries to access an internal state they add a getter to something that shouldn't exist in the first place. I say that if we want to keep this test, just turn into a unit test in package `cardinal` so it can do this abstraction boundaries violation it needs to in the first place and just take the L on it until we untangle the tight coupling.

Reply:
just want to add that this isn't exclusive to `AddEVMTransaction`. It's signature is 1:1 with `AddTransaction`, except it has an extra parameter for the EVM tx source hash.

Reply:
yeah i think both is an issue, i'm just using it as an example becauss it's the first thing i saw in the test mentioned above

Reply:
@smsunarto I agree with your comment on testing. But throwing in huge test refactors along with this change in a single PR makes it hard to read and really stressful for everyone. So my proposal is to do it pieces. Leave this `GetMessageFromWorld` so I can get the refactor in along with good tests to validate it then, remove GetMessageFromWorld and fix the tests in a later endeavor. I can create a ticket and put in a todo comment. 

I'm not sure I understand the part of GetMessage

GetMessage is the primary motivation behind this PR. It's so that users don't have to refer to a global. 

No global value: 
 
 `FooTx.Each(wCtx, ...)` 
 
 instead users do 
 
 `GetMessage[FooIn, FooOut](wCtx)`
 
 This was objectively the primary purpose I was trying to fulfill for this PR. Let me know if that wasn't what you had in mind. This was what I thought was the goal as explained in Tylers description here: https://linear.app/arguslabs/issue/WORLD-840/message-api-refactor

Reply:
i'm not proposing a complete test refactor, you have exactly 1 function that uses GetMessageFormWorld, just rip that one test into a separate test file that is in the cardinal package.

Reply:
left follow up comment on GetMessage below. copying it here for clarity

> nvm, i understand what you are trying to do now, you still can't use message.MessageType though, I would make this private so you don't shoot yourself in the foot. Instead, you need to use types.Message.
>
>I would also move this to cardinal.go together with GetComponent